docs/sql-migration-guide.md

```diff
@@ -23,8 +23,11 @@ license: |
 {:toc}
 
 ## Upgrading from Spark SQL 3.0 to 3.1
+
   - Since Spark 3.1, grouping_id() returns long values. In Spark version 3.0 and earlier, this function returns int values. To restore the behavior before Spark 3.0, you can set `spark.sql.legacy.integerGroupingId` to `true`.
 
+  - Since Spark 3.1, the `formatted` query explain mode is used in the events (e.g., `SparkListenerSQLExecutionStart` and `SparkListenerSQLAdaptiveExecutionUpdate`) of SQL listeners. To restore the behavior before Spark 3.0, you can set `spark.sql.queryDescriptionModeInListeners` to `extended`.
```

Comment:
I think it is too internal to talk about the SQL listener and events like `SparkListenerSQLExecutionStart` here.
How about 
```
SQL UI data adopts the `formatted` mode for the query plan explain results.
```